### PR TITLE
B3 app manager

### PR DESCRIPTION
**Pull request by orangejenny**
Interdependent with https://github.com/dimagi/Vellum/pull/590

If reviewing commit-by-commit, be aware that a few chunks of this have already been cherry-picked and merged to master (https://github.com/dimagi/commcare-hq/pull/9872, https://github.com/dimagi/commcare-hq/pull/9900, https://github.com/dimagi/commcare-hq/pull/9901, https://github.com/dimagi/commcare-hq/pull/9910).

code buddy @TylerSheffels 
B3 context @biyeun 
app manager context @dannyroberts / @snopoke 
fyi @dimagi/product 

Known issues to be fixed before this is merged:
- [x] B3 version or alternative for `alert_user` (see TODO in detail-screen-config.js)
- [x] Tooltips don't work (e.g., hovering over the pencil icons in the menu)
- [x] Vellum full-screen mode doesn't cover top menu (fixed by https://github.com/dimagi/commcare-hq/pull/9932)
- [x] In the modal to edit graphs, the autocompletes for property names aren't working
- [x] Footer isn't behaving quite right (have to scroll to see the bottom half of it)
- [x] Label under form settings > logic is misplaced when feature flag "Allow disabling Form Builder per form" is on
- [x] App Settings > Custom Properties hasn't been migrated

QA ticket: http://manage.dimagi.com/default.asp?204495

app settings
1 settings

case list
2 case list

case list advanced features
3 case list advanced

case detail
4 case detail

form view > case management (basic module)
5 form case mgmt basic

form view > case management (careplan module)
6 form case mgmt careplan

form view > case management (advanced module)
7 form case mgmt advanced

form view > visit scheduler
8 form visit scheduler

deploy page
9 deploy

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/app_manager/templatetags/xforms_extras.py, line 7 - E501 line too long (131 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 51 - E501 line too long (164 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 58 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 59 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 62 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/views/view_generic.py, line 42 - E302 expected 2 blank lines, found 1
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 244 - Bad line breaking before '&&'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 245 - Bad line breaking before '&&'.

**Comment by benrudolph** [comment]
holy :hankey:! this is monstrous! :raised_hands: :raised_hands: :raised_hands:

**Comment by snopoke** [comment]
Indeed a monter! :ocean: :whale2:

**Comment by czue** [comment]
wow! amazing to see this happening! :dragon:

**Comment by biyeun** [comment]
will take a pass after gym. but :+1: SO HAPPY.

File: corehq/apps/app_manager/static/app_manager/json/commcare-app-settings.yaml [hunk 1]
```diff
@@ -146,7 +146,7 @@
   disabled_txt: Oops! This setting has been discontinued. We recommend you change this setting to Version 2, and it will no longer appear on your settings page.
 
 - id: auto_gps_capture
-  name: 'Auto Capture Location (all forms)'
+  name: 'Auto Capture Location<br>(all forms)'
```
Comment (biyeun) [inline corehq/apps/app_manager/static/app_manager/json/commcare-app-settings.yaml]: `<br />`

File: corehq/apps/app_manager/templates/app_manager/app_view.html [hunk 2]
```diff
@@ -295,38 +288,43 @@
...
-            {% trans "The passwords don't match. Please try again." %}
-        </div>
-        <div class="alert alert-danger password-empty hide">
-            {% trans "Please enter a password or cancel." %}
-        </div>
-        <div class="control-group password">
-            <label class="control-label">{% trans "Password" %}</label>
-            <div class="controls">
-                <input type="password" class="form-control" />
+<<<<<<< HEAD
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/app_view.html]: `<<<<<HEAD`

**Comment by dannyroberts** [comment]
:open_mouth: Incredible! :rainbow:

File: corehq/apps/app_manager/templates/app_manager/apps_base.html [hunk 3]
```diff
@@ -125,18 +133,25 @@
...
-            </a>
-            <a href="#" class="btn" data-dismiss="modal">
-                {% trans "No thanks, get me out of here" %}
-            </a>
+    <!-- This will appear on page load, so don't use any animation (normally controlled by .fade) -->
+    <div id="deleted-app-modal" class="modal">
+        <div class="modal-dialog">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h3 class="modal-title">{% trans 'Oops! This application was deleted.' %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/apps_base.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/app_view.html [hunk 4]
```diff
@@ -295,38 +288,43 @@
...
+<div class="modal fade" id="password-setter">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+                <h3 class="modal-title">{% trans "Set Password" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/app_view.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/odk_install.html [hunk 5]
```diff
@@ -1,28 +1,32 @@
...
-</div>
-<div class="modal-footer">
-    <a href="#" class="btn btn-default" data-dismiss="modal">Close</a>
+<div class="modal-dialog modal-lg">
+    <div class="modal-content">
+        <div class="modal-header">
+            <button type="button" class="close" data-dismiss="modal">
+                <span aria-hidden="true">&times;</span>
+            </button>
+            <h3 class="modal-title">
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/odk_install.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/case_config.html [hunk 6]
```diff
@@ -3,16 +3,22 @@
...
-        <p>{% trans "Are you sure you want to remove this subcase?" %}</p>
-    </div>
-    <div class="modal-footer">
-        <a href="#" class="btn" data-dismiss="modal">{% trans "Cancel" %}</a>
-        <a class="btn btn-danger" href="#" data-bind="click: $parent.removeSubCase" data-dismiss="modal">{% trans "Remove Subcase" %}</a>
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <a href="#" class="close" data-dismiss="modal">×</a>
+                <h3 class="modal-title">{% trans "Remove Subcase?" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/case_config.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/case_config_advanced.html [hunk 7]
```diff
@@ -3,16 +3,22 @@
...
-    <div class="modal-footer">
-        <a href="#" class="btn" data-dismiss="modal">{% trans "Cancel" %}</a>
-        <a class="btn btn-danger" href="#" data-bind="click: $parent.removeFormAction" data-dismiss="modal">{% trans "Remove Action" %}</a>
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+                <h3 class="modal-title">{% trans "Remove Form Action?" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/case_config_advanced.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/download_async_modal.html [hunk 8]
```diff
@@ -20,12 +20,14 @@
     <div class="modal-dialog">
         <div class="modal-content">
             <div class="modal-header">
-                <a class="close" data-dismiss="modal">&times;</a>
-                <h3>{% trans "Downloading Your File" %}</h3>
+                <button type="button" class="close" data-dismiss="modal">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+                <h3 class="modal-title">{% trans "Downloading Your File" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/download_async_modal.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html [hunk 9]
```diff
@@ -2,104 +2,118 @@
...
+                <a href="#upload-xform" data-toggle="modal">
+                    <i class="fa fa-arrow-up"></i>
+                    {% trans "Upload" %}
+                </a>
+                <div class="modal fade" id="upload-xform">
+                    <div class="modal-dialog">
+                        <div class="modal-content">
+                            <div class="modal-header">
+                                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                                <h3 class="modal-title">{% trans "Upload XForm" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html [hunk 10]
```diff
@@ -2,104 +2,118 @@
...
+                    {% trans "View" %}
+                </a>
+                <div class="xml-source modal fade">
+                    <div class="modal-dialog">
+                        <div class="modal-content">
+                            <form action="{% url "corehq.apps.app_manager.views.edit_form_attr" domain app.id form.get_unique_id 'xform' %}" method="POST">
+                            {% csrf_token %}
+                            <div class="modal-header">
+                                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                                <h3 class="modal-title source-readonly">{% trans "XML Source" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html [hunk 11]
```diff
@@ -2,104 +2,118 @@
...
+                                <h3 class="modal-title source-edit">{% trans "Edit XML Source" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/graph_configuration_modal.html [hunk 12]
```diff
@@ -1,199 +1,218 @@
...
-                        <li><a>Annotations</a></li>
-                    </ul>
+    <div class="modal fade graph-config">
+        <div class="modal-dialog modal-lg">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                    <h3 class="modal-title" data-bind="text: 'Editing ' + graphDisplayName()"></h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/graph_configuration_modal.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/nav_menu_media_modals.html [hunk 13]
```diff
@@ -1,73 +1,81 @@
...
-            </label>
-            <div class="controls">
-                <div class="hqm-select-files-container">
-                    <button class="hqm-select btn btn-primary" role="button">{% trans 'Select' %}</button>
+<div class="modal fade" id="{{ multimedia.upload_managers.icon.slug }}">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                <h3 class="modal-title">{% trans 'Upload Icon Image' %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/nav_menu_media_modals.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/nav_menu_media_modals.html [hunk 14]
```diff
@@ -1,73 +1,81 @@
...
-                <div class="hqm-select-files-container">
-                    <button class="hqm-select btn btn-primary" role="button">{% trans 'Select' %}</button>
-                </div>
-                <div class="hqm-queue">
+<div class="modal fade" id="{{ multimedia.upload_managers.audio.slug }}">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                <h3 class="modal-title">{% trans 'Upload Menu Audio' %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/nav_menu_media_modals.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/new_careplan_module.html [hunk 15]
```diff
@@ -1,34 +1,39 @@
...
-                            {% endif %}
-                        {% endfor %}
-                        </select>
-                    </div>
+    <div id="careplan-module-modal" class="modal fade">
+        <div class="modal-dialog">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                    <h3 class="modal-title">{% trans "New Care Plan Module" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/new_careplan_module.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/partials/releases_deploy_modal.html [hunk 16]
```diff
@@ -1,239 +1,262 @@
...
-                <a class="accordion-toggle">
-                    {% trans "Download to Android" %}
-                </a>
+<div class="modal-dialog">
+    <div class="modal-content">
+        <div class="modal-header">
+            <button type="button" class="close" data-dismiss="modal">
+                <span aria-hidden="true">&times;</span>
+            </button>
+            <h3 class="modal-title">{% trans "Deploying version" %} <span data-bind="text: version"></span></h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/partials/releases_deploy_modal.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/releases.html [hunk 17]
```diff
@@ -308,31 +308,43 @@ <h6 data-bind="if: !built_with.signed()">{% trans "Unsigned Jar" %}</h6>
...
-            <div class="modal-footer">
-                <a href="#" class="btn" data-dismiss="modal">Cancel</a>
-                <a class="btn btn-danger" href="#" data-bind="click: $root.deleteSavedApp" data-dismiss="modal">{% trans "Delete Version" %}</a>
+            <div class="modal-dialog">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal">
+                            <span aria-hidden="true">&times;</span>
+                        </button>
+                        <h3 class="modal-title">{% trans "Delete Version?" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/releases.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/releases.html [hunk 18]
```diff
@@ -308,31 +308,43 @@ <h6 data-bind="if: !built_with.signed()">{% trans "Unsigned Jar" %}</h6>
...
-            <div class="modal-footer">
-                <a href="#" class="btn" data-dismiss="modal">Cancel</a>
-                <a href="#" class="btn btn-danger" data-bind="click: $root.revertSavedApp" data-dismiss="modal">{% trans "Revert" %}</a>
+            <div class="modal-dialog">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal">
+                            <span aria-hidden="true">&times;</span>
+                        </button>
+                        <h3 class="modal-title">{% trans "Revert to Version?" %}</h3>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/releases.html]: `h4`

File: corehq/apps/style/templates/style/bootstrap3/partials/key_value_mapping.html [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+
+<div id="key_value_mapping_modal" class="hide">
+    <div class="modal fade hq-enum-modal">
+        <div class="modal-dialog">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                    <h3 class="modal-title" data-bind="text: $data.modalTitle"></h3>
```
Comment (biyeun) [inline corehq/apps/style/templates/style/bootstrap3/partials/key_value_mapping.html]: `h4`

File: corehq/apps/app_manager/templates/app_manager/import_app.html [hunk 20]
```diff
@@ -32,29 +31,42 @@
...
-            </tr>
-            <tr>
-                <th>Source:</th>
-                <td><textarea name="source" data-bind="value: source"></textarea></td>
-            </tr>
-        </table>
-        <a href="#" class="historyBack">No, take me back.</a>
-        <button class="btn btn-primary" type="submit">{% trans 'Yes, import application' %}</button>
+            <div class="form-group">
+                <label class="col-lg-2 control-label">Name</label>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/import_app.html]: what about `md` and `sm` views? most of our users fall in that category...

File: corehq/apps/app_manager/templates/app_manager/module_view.html [hunk 21]
```diff
@@ -210,6 +206,7 @@
     {% endfor %}
     {% endblock %}{# endblock tab-headers #}
 </ul>
+<br>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/module_view.html]: `<br />`

File: corehq/apps/app_manager/templates/app_manager/module_view.html [hunk 22]
```diff
@@ -217,20 +214,20 @@
             {% csrf_token %}
             <div class="save-button-holder clearfix"></div>
             <fieldset>
-                <div class="control-group">
-                    <label class="control-label">{% trans "Module Name" %}</label>
-                    <div class="controls">
-                        {% input_trans module.name langs %}
+                <div class="form-group">
+                    <label class="col-lg-2 control-label">{% trans "Module Name" %}</label>
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/module_view.html]: don't forget about `col-sm` and `col-md` too

File: corehq/apps/app_manager/templates/app_manager/import_app.html (hunk 20 above)
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/import_app.html]: don't forget about `col-sm` and `col-md` too

File: corehq/apps/app_manager/templates/app_manager/module_view.html (hunk 22 above)
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/module_view.html]: don't forget about `col-sm` and `col-md` too

File: corehq/apps/app_manager/templates/app_manager/module_view_advanced.html [hunk 23]
```diff
@@ -48,16 +48,16 @@
 {% endblock %}
 
 {% block case_management_advanced_fields %}{{ block.super }}
-    <div class="control-group">
-        <label class="control-label">
+    <div class="form-group">
+        <label class="control-label col-lg-2">
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/module_view_advanced.html]: don't forget about `col-sm` and `col-md` too

File: corehq/apps/app_manager/templates/app_manager/module_view_careplan.html [hunk 24]
```diff
@@ -20,18 +20,19 @@
 {% endblock %}
 
 {% block settings_fields %}
-    <div id="detail-screen-parent" class="control-group">
-        <label class="control-label">{% trans "Attached to" %}</label>
-        <div class="controls">
-            <select name="parent_module" data-bind="optstr: moduleOptions, value: moduleId"></select>
+    <div id="detail-screen-parent" class="form-group">
+        <label class="control-label col-lg-2">{% trans "Attached to" %}</label>
+        <div class="col-lg-4">
```
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/module_view_careplan.html]: don't forget about `col-sm` and `col-md` too

**Comment by biyeun** [comment]
Started leaving a few comments, but just want to make sure that there aren't any solo `col-lg` styles without `col-sm` and `col-md` because most of our users don't have large screens!

Also, as a todo for the future, (@dannyroberts) would be good to component-alize (knockout 3) modals. For the other b3 pages i've been using the following in the `modal-header` for consistency (note `h4` vs `h3` semantic usage).:
```<div class="modal-header">
    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
    <h4 class="modal-title">{{TITLE}}</h4>
</div>
```
If this were a component life would be so much easier *sigh*. :crying_cat_face:

File: corehq/apps/hqwebapp/static/hqwebapp/js/ui-element.js [hunk 25]
```diff
@@ -93,9 +93,9 @@ var uiElement;
             var translated = langcodeTag.translate_delim(value);
             this.ui.find('.lang-text').remove();
             if (translated.lang) {
+                this.ui.css("position", "relative");
                 var langcode_button = langcodeTag.button_tag(
-                    $('<a href="#" class="btn btn-inverse btn-mini lang-text"'
-                      + ' style="color:#ffffff; text-decoration: none;" />'),
+                    $('<a href="#" class="btn btn-info btn-xs btn-mini lang-text" style="position: absolute; top: 6px; right: 6px;" />'),
```
Comment (biyeun) [inline corehq/apps/hqwebapp/static/hqwebapp/js/ui-element.js]: guessing this is still b2 supported? `btn-mini`

File: corehq/apps/style/templates/style/bootstrap3/partials/key_value_mapping.html [hunk 26]
```diff
@@ -0,0 +1,99 @@
...
+                    <div class="col-sm-1">
+                        <a href="#" data-bind="click: $parent.removeItem" class="btn btn-danger">
+                            <i class="icon-white fa fa-remove"></i>
+                        </a>
+                    </div>
+                </div>
+            </div>
+        </fieldset>
+
+        <div class="control-group">
```
Comment (biyeun) [inline corehq/apps/style/templates/style/bootstrap3/partials/key_value_mapping.html]: `form-group`

**Comment by biyeun** [comment]
pt 1 review complete

File: corehq/apps/hqwebapp/static/hqwebapp/js/ui-element.js (hunk 25 above)
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/hqwebapp/js/ui-element.js]: I didn't check where all ui element is used, but looking now, does look like it's mostly app manager, with one reference in `corehq/apps/groups/templates/groups/group_members.html` (not sure if that's B2 or B3). I'm inclined to leave it B2-compatible until we're all switched over and then do a final sweep for B2 classes. There are other js files that'll need that kind of cleanup anyway (the multimedia popup comes to mind).

**Comment by orangejenny** [comment]
@biyeun  thanks for going through all this.

> just want to make sure that there aren't any solo col-lg styles without col-sm and col-md because most of our users don't have large screens!

Ah, there's a ton of these, I'll take a pass at fixing them. What's our best practice? Including `sm`, `md`, and `lg` everywhere? (Talking about places where we don't care about varying the proportions on different screen sizes, so it'd be `col-lg-2 col-md-2 col-sm-2`.)

> would be good to component-alize (knockout 3) modals

Strongly agree! Doing this migration got me excited about looking into knockout's components. Upgrading to knockout 3 should be a priority after the B3 migration is over (for the hopefully few areas where knockout upgrade wasn't done along with B3 upgrade).

File: corehq/apps/app_manager/templates/app_manager/apps_base.html (hunk 3 above)
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/apps_base.html]: yeah, i did h3 everywhere, it was what the migration guide happened to have. updated migration guide, will fix the PR.

**Comment by biyeun** [comment]
@orangejenny then the best option is to default to the smaller screen so instead of `lg` use `sm`

**Comment by orangejenny** [comment]
Cool, will do.

File: corehq/apps/app_manager/static/app_manager/js/case-config-ui-advanced.js [hunk 27]
```diff
@@ -152,18 +152,16 @@ var AdvancedCase = (function () {
...
-                if (index) {
-                    options.active = index;
-                }
-                if (!type || type === 'open') {
-                    $('#case-open-accordion').accordion("destroy").accordion(options);
-                }
-                if (!type || type === 'load') {
-                    $('#case-load-accordion').accordion("destroy").accordion(options);
-                }
-            });
```
Comment (snopoke) [inline corehq/apps/app_manager/static/app_manager/js/case-config-ui-advanced.js]: I'd be quite happy to get rid of these accordion's completely

File: corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js [hunk 28]
```diff
@@ -494,7 +497,9 @@ var DetailScreenConfig = (function () {
                 hasNodeset: false,
                 nodeset: "",
             };
-            _.defaults(this.original, tabDefaults);
+            _.each(_.keys(tabDefaults), function(key) {
+                that.original[key] = that.original[key] || tabDefaults[key];
+            });
```
Comment (snopoke) [inline corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js]: Why the change from defaults

File: corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js [hunk 29]
```diff
@@ -101,9 +101,7 @@ ko.bindingHandlers.casePropertyTypeahead = {
 ko.bindingHandlers.accordion = {
     init: function(element, valueAccessor) {
         var options = valueAccessor() || {};
-        setTimeout(function() {
-            $(element).accordion(options);
-        }, 0);
```
Comment (snopoke) [inline corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js]: I see a few accordion related changes but assume you've tested them

File: corehq/apps/app_manager/templates/app_manager/app_view.html (hunk 2 above)
Comment (snopoke) [inline corehq/apps/app_manager/templates/app_manager/app_view.html]: :suspect:

File: corehq/apps/app_manager/templates/app_manager/form_view_base.html [hunk 30]
```diff
@@ -339,30 +297,17 @@
...
-                            {% block case_management_content %}
-                                {% include 'app_manager/partials/case_config.html' %}
-                            {% endblock %}
-                        </div>
+                    <div class="help-block">
+                        Cases give you a way to track patients, farms, etc. over time. You can choose to save 
+                        data from a form to the case, which will store the data locally on the phone to use later.
+                    </div>
+                    <div class="casexml" id="casexml_home">
+                    {% if app.application_version != '1.0' %}
```
Comment (snopoke) [inline corehq/apps/app_manager/templates/app_manager/form_view_base.html]: Removed support for `1.0` apps. Getting closer to :skull:

File: corehq/apps/app_manager/templates/app_manager/languages.html [hunk 31]
```diff
@@ -34,26 +35,25 @@
...
-                            For Java Phone (Nokia) refer to
+                            For Java Phone (Nokia), refer to
                             <a target="_blank" href="https://github.com/dimagi/javarosa/blob/master/j2me/shared-resources/resources/messages_default.txt">JavaRosa translations</a>
                             and <a target="_blank" href="https://github.com/dimagi/commcare/blob/master/application/resources/messages_cc_default.txt">CommCare translations</a>.
                             {% endblocktrans %}
                         </p>
                         <p class="help-block">
                             {% blocktrans %}
-                            For Android refer to
+                            For Android, refer to
```
Comment (snopoke) [inline corehq/apps/app_manager/templates/app_manager/languages.html]: Even grammar corrections :angel:

File: corehq/apps/app_manager/static/app_manager/js/case-config-ui-advanced.js (hunk 27 above)
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/case-config-ui-advanced.js]: It's on my todo list :)  Hoping to squeeze it into this PR.

File: corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js (hunk 28 above)
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js]: Underscore 1.5 updated defaults to only override undefined values. Previously it had been all falsy values, which is what this code needs (it's largely overwriting nulls, not undefined).

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/app_manager/templatetags/xforms_extras.py, line 7 - E501 line too long (131 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 51 - E501 line too long (164 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 58 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 59 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 62 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/views/view_generic.py, line 42 - E302 expected 2 blank lines, found 1
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 244 - Bad line breaking before '&&'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 245 - Bad line breaking before '&&'.

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/app_manager/templatetags/xforms_extras.py, line 7 - E501 line too long (131 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 51 - E501 line too long (164 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 58 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 59 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 62 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/views/view_generic.py, line 42 - E302 expected 2 blank lines, found 1
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 244 - Bad line breaking before '&&'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 245 - Bad line breaking before '&&'.

File: corehq/apps/app_manager/templates/app_manager/form_view_base.html (hunk 30 above)
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/form_view_base.html]: Awesome - can I remove all 1.0-related code in app manager, then?

File: corehq/apps/app_manager/templatetags/xforms_extras.py [hunk 32]
```diff
@@ -45,17 +48,25 @@ def html_name(name):
...
             if langs and lang == langs[0]:
                 return template % {"value": name[lang], "placeholder": ""}
             else:
                 return template % {"value": "", "placeholder": name[lang]} + \
-                       LANG_BUTTON % {"lang": lang, "extra_class": " langcode-input"}
+                    LANG_BUTTON % {
+                        "lang": lang,
+                        "extra_class": " langcode-input",
+                        "extra_style": "position: absolute; top: 6px; right: 15px"
+                    }
```
Comment (dimagimon) [inline corehq/apps/app_manager/templatetags/xforms_extras.py]: E123 closing bracket does not match indentation of opening bracket's line

File: corehq/apps/app_manager/templates/app_manager/form_view_base.html (hunk 30 above)
Comment (snopoke) [inline corehq/apps/app_manager/templates/app_manager/form_view_base.html]: Not quite yet, but hopefully in the next month or so.

**Comment by orangejenny** [comment]
killing some intermediate builds

File: corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js (hunk 29 above)
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js]: I didn't thoroughly dig through why all of them were necessary (in particular, swapping `load` and `save` in `case-config-ui-advanced.js` felt weird) but yeah, I did test the accordion.

**Comment by orangejenny** [comment]
QA passed, this is good to merge once https://github.com/dimagi/Vellum/pull/590 is merged. Would like to deploy today.

**Comment by dannyroberts** [comment]
Woohoo! :star2:

**Comment by czue** [comment]
beastly! :bear: :wolf: :tiger:
